### PR TITLE
feat(channels): new channel search form

### LINKED ISSUE #1571: I should see a summary of my final channel configuration before I submit the new channel form
## Detailed Description

After filling in the new channel form, I should see a summary of the channel config before finally submitting it so that I can verify everything is how I want it to be. Similar to what the user sees when making payments.

image

## Context

New Channel UI

### PR DESCRIPTION
**Pull request by mrfelton**
## Description:

Build new channel search form

## Motivation and Context:

Resolves #1571

## How Has This Been Tested?

Manually

## Screenshots (if appropriate):

image

## Types of changes:

New Feature

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have read the _CONTRIBUTING_ document.
- [ ] I have added tests to cover my changes where needed.
- [ ] All new and existing tests passed.
- [x] My commits have been squashed into a concise set of changes.

### CODE REVIEW

File: app/reducers/contactsform.js [hunk 1]
```diff
@@ -201,6 +201,17 @@ contactFormSelectors.filteredNetworkNodes = createSelector(
   }
 )
 
+contactFormSelectors.isSearchValidNodeAddress = createSelector(
+  searchQuerySelector,
+  searchQuery => {
+    if (!searchQuery) {
+      return false
+    }
+    const [pubkey, host] = searchQuery.split('@')
```
Comment (korhaliv) [inline app/reducers/contactsform.js]: Could be a little more efficient if you first check for `searchQuery.length` to be equal 2

File: app/components/Channels/ChannelNodeSearch.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    {filteredNetworkNodes.map(node => {
+      return (
+        <React.Fragment key={node.pub_key}>
+          <Flex justifyContent="space-between" alignItems="center" py={2} pr={2}>
+            <Box mr={3}>
+              {node.alias.length > 0 ? (
+                <>
+                  <ClippedText>{node.alias.trim()}</ClippedText>
+                  <Text color="gray" fontSize="s">
+                    <Truncate text={node.pub_key} maxlen={25} />
```
Comment (korhaliv) [inline app/components/Channels/ChannelNodeSearch.js]: What's the difference between `ClippedText` and `Truncate` ? Sounds similar judging from the names

File: app/reducers/contactsform.js (hunk 1 above)
Comment (mrfelton) [reply app/reducers/contactsform.js]: I don't think that would have the same effect. I specifically want to ensure that we have something before and after an `@` to ensure that what we have looks like a node address.

File: app/components/Channels/ChannelNodeSearch.js (hunk 2 above)
Comment (mrfelton) [reply app/components/Channels/ChannelNodeSearch.js]: `ClippedText` chips off long text with an ellipsis. `Truncate` truncates the middle of a string.

File: app/reducers/contactsform.js (hunk 1 above)
Comment (korhaliv) [reply app/reducers/contactsform.js]: Yes, that's why I said first. So it's an addition. You don't need to split if length != 2 right?

Comment (mrfelton) [reply app/reducers/contactsform.js]: Got it. Updated.

**Pull request merged**